pkg/reconciler/v1alpha1/revision/resources/deploy.go

```diff
@@ -114,7 +115,10 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, observab
 	// Adding or removing an overwritten corev1.Container field here? Don't forget to
 	// update the validations in pkg/webhook.validateContainer.
 	userContainer.Name = UserContainerName
-	userContainer.Resources = userResources
+
+	if equality.Semantic.DeepEqual(userContainer.Resources, corev1.ResourceRequirements{}) {
```

Comment:
Should this be !DeepEqual?

Reply:
My intention is replace the resources with our recommend only if the user didn't specify them. 

Maybe I should rename `userResources` to  `defaultUserResources` to make this clearer. 

https://github.com/knative/serving/blob/master/pkg/reconciler/v1alpha1/revision/resources/deploy.go#L76-L80

Reply:
Aha, I hadn't realized that this was an unconditional override already. Thanks!